azure-configurationservice/azure/configuration/azure_configuration_client.py

```diff
@@ -0,0 +1,254 @@
+        """
+        Example
+
+        .. code-block:: python
+
+            from datetime import datetime, timedelta
+
+            accept_date_time = datetime.today() + timedelta(days=-1)
+
+            all_listed = client.list_configuration_settings()
+            for item in all_listed:
+                pass  # do something
+
+            filtered_listed = client.list_configuration_settings(
+                labels=["*Labe*"], keys=["*Ke*"], accept_date_time=accept_date_time
+            )
+            for item in filtered_listed:
+                pass  # do something
+        """
+        return super(AzureConfigurationClient, self).list_configuration_settings(
+            labels, keys, accept_date_time, fields, **kwargs
+        )
+
+    def get_configuration_setting(
+        self, key, label=None, accept_date_time=None, **kwargs
+    ):  # type: (str, str, datetime, dict) -> Paged
+        """
+        Example
+
+        .. code-block:: python
+
+            fetched_config_setting = client.get_configuration_setting(
+                key="MyKey", label="MyLabel"
+            )
+        """
+        return super(AzureConfigurationClient, self).get_configuration_setting(
+            key, label=label, accept_date_time=accept_date_time, **kwargs
+        )
+
+    def add_configuration_setting(self, configuration_setting, **kwargs):
+        # type: (ConfigurationSetting, dict) -> ConfigurationSetting
+        """
+        Example
+
+        .. code-block:: python
+
+            config_setting = ConfigurationSetting(
+                key="MyKey",
+                label="MyLabel",
+                value="my value",
+                content_type="my content type",
+                tags={"my tag": "my tag value"}
+            )
+            added_config_setting = client.add_configuration_setting(config_setting)
+        """
+        return super(AzureConfigurationClient, self).add_configuration_setting(configuration_setting, **kwargs)
+
+    def update_configuration_setting(
+        self,
+        key,
+        value=None,
+        content_type=None,
+        tags=None,
+        label=None,
+        etag=None,
+        **kwargs
+    ):  # type: (str, str, str, dict, str, str, dict) -> ConfigurationSetting
+        """
+        Example
+
+        .. code-block:: python
+
+            updated_kv = client.update_configuration_setting(
+                key="MyKey",
+                label="MyLabel",
+                value="my updated value",
+                content_type=None,  # None means not to update it
+                tags={"my updated tag": "my updated tag value"}
+            )
+        """
+        custom_headers = AzureConfigurationClientAbstract.prep_update_configuration_setting(
+            key, etag, **kwargs
+        )
+
+        current_configuration_setting = self.get_configuration_setting(key, label)
+        if value is not None:
+            current_configuration_setting.value = value
+        if content_type is not None:
```

Comment:
How would I remove the content_type? Pass in an empty string?

Reply:
Yes. I think this is the original design. None means no change. If None doesn't mean no change, we need to have a different sentinel that means no change